packages/generators/add-generator.ts

```diff
@@ -151,13 +104,29 @@ export default class AddGenerator extends Generator {
 							);
 					} else {
 						if (action === "topScope") {
-							return this.prompt([Input("topScope", "What do you want to add to topScope?")]).then(
+							return this.prompt(topScopeQuestion).then(
 								(topScopeAnswer: { topScope: string }): void => {
 									this.configuration.config.topScope.push(topScopeAnswer.topScope);
 									done();
 								}
 							);
 						}
+						if (action === "merge") {
+							return this.prompt([mergeFileQuestion])
+							.then((mergeFileAnswer: {
+								mergeFile: string;
+							}) => {
+								const resolvedPath = resolve(process.cwd(), mergeFileAnswer.mergeFile);
```

Comment:
You'll need to parse this yourself. Extract this logic into a function `parseMergeConfig`, where you read the config, the `webpackOptions` is the module export of the config itself, you already have the filename from `basename(resolvedPath)` . I don't think that merge actually works like this. Merge takes a config name and attaches the name to a function. 

What you will need to do, is to make merge take in either `<Array>[name, path]` and add a require on top if the require is not in topScope (which I think is the best solution) or just add the name (bad DX).

So what you'll need to do, is to have this in the object of the generator:

```js
{
merge: ['webpackBaseConfig', 'path/to/foo'],
config {...}
}
```

and make the function parseMerge ( https://github.com/webpack/webpack-cli/blob/master/packages/utils/ast-utils.ts#L608-L640 ) accept an array that will convert the first idx to the function parameter, the second one to a topScope property. You could use the existing API to make this happen more easily, and that is to make a require statement based on the indices and add that to topScope before running the transform. This will leave you with the same API as it is, but providing support for `merge` in add.

Reply:
Nice solution.
Or we can just take up the path to the config file from user. Internally, add `webpack-merge` as dependency for the user and use it to scaffold a config that will use `webpack-merge`.
WDYT?

Reply:
Done @evenstensberg

Reply:
Nice. Have a look at the comment further down